File: egolint/tests/fixtures/checkstyle/GreetingService.java
package fixtures.checkstyle;

import java.util.Objects;

/**
 * Creates personalized greeting messages.
 */
public final class GreetingService {

    private final String prefix;

    /**
     * Creates a greeting service.
     *
     * @param prefix greeting prefix
     */
    public GreetingService(String prefix) {
        this.prefix = Objects.requireNonNull(prefix);
    }

    /**
     * Creates a greeting for a supplied name.
     *
     * @param name greeting recipient
     * @return formatted greeting
     */
    public String createGreeting(String name) {
        if (name == null || name.isBlank()) {
            return prefix;
        }

        return "%s, %s!".formatted(prefix, name.trim());
    }
}

File: egolint/tests/fixtures/revive/greeting_service.go
// Package revive provides source used to validate Revive integration.
package revive

import (
	"fmt"
	"strings"
)

// GreetingService creates personalized greeting messages.
type GreetingService struct {
	prefix string
}

// NewGreetingService creates a greeting service with a normalized prefix.
func NewGreetingService(prefix string) GreetingService {
	return GreetingService{
		prefix: strings.TrimSpace(prefix),
	}
}

// CreateGreeting creates a greeting for the supplied name.
func (service GreetingService) CreateGreeting(name string) string {
	normalizedName := strings.TrimSpace(name)
	if normalizedName == "" {
		return service.prefix
	}

	return fmt.Sprintf("%s, %s!", service.prefix, normalizedName)
}
